src/test/java/gov/usgs/wma/waterdata/GlobalDefaultExceptionHandlerTest.java

```diff
@@ -75,10 +79,14 @@ public void handleMultilineHttpMessageNotReadableException() throws IOException
 
 	@Test
 	public void handleUncaughtExceptionTest() throws IOException {
+		Logger root = (Logger)LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME);
```

Comment:
What is the purpose of this? Seems risky to mess with logging levels just to keep a unit test log clean.

Reply:
Oh, I disagree. This is a good thing that has little risk and high reward. When tests fail with exceptions in the logs, it is noise to locating the actual cause of failed tests. I planned to add this to all tests that log an exception; I did not notice any others. The risk of noise is worse than the risk of developer time to look into an error that is suppose to happen. Finally, if it ends up causing some side effect (I presume is the risk you refer) then we could update the test with any improvements that would address it. The risk is also isolated to tests. It does not effect the release. I considered wrapping it in the try-finally as I have done in the past but spring-boot is restarted for each testing class. Logging should be recreated in the bootstrap process. A new log is a new root setting. That all said: It took me longer to explain it than to just do it. So, if you like a try-finally wrapper, I am game.

Reply:
Do not go with a try catch. 
I would argue that the full complete test would actually check the log to guarantee that the log got written...

Setting root to off could also suppress any logs that could help in diagnosing a failure in this test.

The only way to guarantee a reload of the spring context is to use the @DirtiesContext and @TestExecutionListeners(DirtiesContextTestExecutionListener.class)

Reply:
I fully understand the ramifications of turning off the logs. In fact, what you seem to be concerned about is exactly want I want to happen. I want clean logs. No exceptions when no error really happened. It make it hard to know what is going on when things fail. We should have a discussion about this to come to a resolution. I am disinclined to turn the exception log back on and inclined to apply this to any other test logging I see in the future that is just noise. Well, that I say test only apply to the brute force means to clean the log for the test logging. 

Yes, it makes sense to test that the log wrote the exception. That test should redirect the logging from the console (or test logging destination) entirely, and in that capture, check that expected was logged. I have actually done that in the past. It is more involved than just quieting them for a test that did not check the log in the first place. Because this test did not check for the logging, I quieted it. Had it checked for the logging, then I would have resolved it another way. It might not be idea but it is fine.

Reply:
I added the dirty context annotations.